polygon/dex/trades/insert_clipper_coves.sql

```diff
@@ -0,0 +1,132 @@
+        block_time,
+        token_a_symbol,
+        token_b_symbol,
+        token_a_amount,
+        token_b_amount,
+        project,
+        version,
+        category,
+        trader_a,
+        trader_b,
+        token_a_amount_raw,
+        token_b_amount_raw,
+        usd_amount,
+        token_a_address,
+        token_b_address,
+        exchange_contract_address,
+        tx_hash,
+        tx_from,
+        tx_to,
+        trace_address,
+        evt_index,
+        trade_id
+    )
+    SELECT
+        dexs.block_time,
+        erc20a.symbol AS token_a_symbol,
+        erc20b.symbol AS token_b_symbol,
+        token_a_amount_raw / 10 ^ erc20a.decimals AS token_a_amount,
+        token_b_amount_raw / 10 ^ erc20b.decimals AS token_b_amount,
+        project,
+        version,
+        category,
+        coalesce(trader_a, tx."from") as trader_a, -- subqueries rely on this COALESCE to avoid redundant joins with the transactions table
+        trader_b,
+        token_a_amount_raw,
+        token_b_amount_raw,
+        coalesce(
+            usd_amount,
+            token_a_amount_raw / 10 ^ erc20a.decimals * pa.price,
+            token_b_amount_raw / 10 ^ erc20b.decimals * pb.price
+        ) as usd_amount,
+        token_a_address,
+        token_b_address,
+        exchange_contract_address,
+        tx_hash,
+        tx."from" as tx_from,
+        tx."to" as tx_to,
+        trace_address,
+        evt_index,
+        row_number() OVER (PARTITION BY project, tx_hash, evt_index, trace_address ORDER BY version, category) AS trade_id
+    FROM (
+        -- ClipperCove Polygon
+        SELECT
+            t.evt_block_time AS block_time,
+            'Clipper' AS project,
+            'coves1' AS version,
+            'DEX' AS category,
+            t."recipient" AS trader_a,
+            NULL::bytea AS trader_b,
+            "inAmount" AS token_b_amount_raw,
+            "outAmount" AS token_a_amount_raw,
+            -- Cast raw sigil as MATIC for price purposes
+            CASE WHEN "outAsset" = '\x0000000000000000000000000000000000000000' THEN
+                '\x0000000000000000000000000000000000001010'::bytea ELSE "outAsset"
+            END AS token_a_address,
+            CASE WHEN "inAsset" = '\x0000000000000000000000000000000000000000' THEN
+                '\x0000000000000000000000000000000000001010'::bytea ELSE "inAsset"
+            END AS token_b_address,
+            t.contract_address AS exchange_contract_address,
+            t.evt_tx_hash AS tx_hash,
+            NULL::integer[] AS trace_address,
+            t.evt_index
+        FROM
+            clipper."ClipperCove_evt_CoveSwapped" t
+    ) dexs
+    INNER JOIN polygon.transactions tx
+        ON dexs.tx_hash = tx.hash
+        AND tx.block_time >= start_ts
+        AND tx.block_time < end_ts
+        AND tx.block_number >= start_block
+        AND tx.block_number < end_block
+    LEFT JOIN erc20.tokens erc20a ON erc20a.contract_address = dexs.token_a_address
```

Comment:
i added the tokens from your edited file, however, i believe there may still be some missing from this dex. with missing tokens in erc20.tokens, there can be missing data in amount fields and symbols. you may look to add more tokens to the file using this as reference:
https://dune.com/queries/956351?d=7

in addition to the tokens table entries, may want to look at adding all the tokens to prices file as well, if missing:
https://github.com/duneanalytics/abstractions/blob/master/prices/polygon/coingecko.yaml

we should look to get the best data quality we can prior to initial history backfill. let me know if this all makes sense

Reply:
Thanks. I added a couple more notable tokens and added Coingecko feeds for all of the new entries. The biggest remaining ones are wormhole tokens - making a decision about whether wormhole USDC is "real" USDC is beyond the scope of this PR.

Reply:
I'll add up to the ones with 20 txs. Since Clipper Coves can trade any ERC20 token we won't be able to cover _everything_. The value calculation will just default to 0 - a failsafe - if it's swapping tokens we don't know anything about.

Reply:
yep, fully understand. just double checking. all good!